### PR TITLE
MASTERDIET-31 - Diary

### PR DESCRIPTION
**Pull request by jakubowiczish**

### CODE REVIEW

File: src/main/java/pl/agh/edu/master_diet/core/model/rest/diary/SingleRecentProductResponse.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import pl.agh.edu.master_diet.core.model.shared.MealType;
+import pl.agh.edu.master_diet.core.model.shared.Unit;
+
+import java.time.LocalDateTime;
+
+@Getter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class SingleRecentProductResponse {
```
Comment (jfoder) [inline src/main/java/pl/agh/edu/master_diet/core/model/rest/diary/SingleRecentProductResponse.java:17]: This class isn't used directly as a response, it's only contained by response model. Maybe it should be renamed to something like `SingleRecentProductInfo`?

File: src/main/java/pl/agh/edu/master_diet/service/RecentProductService.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                .mealTime(recentProduct.getMealTime())
+                .recentProductId(recentProduct.getId())
+                .portion(recentProduct.getPortion())
+                .amount(recentProduct.getAmount())
+                .mealType(recentProduct.getMealType())
+                .productName(product.getName())
+                .build();
+    }
+
+
```
Comment (jfoder) [inline src/main/java/pl/agh/edu/master_diet/service/RecentProductService.java:100]: Unnecessary blank lines

File: src/main/java/pl/agh/edu/master_diet/service/converter/RecentProductConverter.java [hunk 3]
```diff
@@ -22,4 +25,30 @@ public BaseProductInfo createFrom(final RecentProduct dto) {
...
+                .amount(parameters.getAmount())
+                .mealType(parameters.getMealType())
+                .mealTime(parameters.getMealTime())
+                .portion(parameters.getPortion())
+                .portionUnit(parameters.getPortionUnit())
+                .product(product)
+                .user(user)
+                .build();
+    }
+
```
Comment (jfoder) [inline src/main/java/pl/agh/edu/master_diet/service/converter/RecentProductConverter.java:53]: Unnecessary blank line

File: src/main/java/pl/agh/edu/master_diet/util/GlobalConstants.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package pl.agh.edu.master_diet.util;
+
+public final class GlobalConstants {
+
+    public static final String GLOBAL_DATE_FORMAT = "dd/MM/yyyy";
+
+    private GlobalConstants() {
+    }
+
```
Comment (jfoder) [inline src/main/java/pl/agh/edu/master_diet/util/GlobalConstants.java:9]: Unnecessary blank line

File: src/main/java/pl/agh/edu/master_diet/controller/RecentProductController.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    public ResponseEntity<DeleteRecentProductsResponse> deleteRecentProducts(
+            @Valid @RequestBody final DeleteRecentProductsRequest request,
+            @CurrentUser final UserPrincipal userPrincipal) {
+
+        log.info("Attempt to delete recent products");
+        return ResponseEntity.ok()
+                .body(recentProductService
+                        .deleteRecentProducts(request.getRecentProductsIds(), userPrincipal.getId()));
+    }
+
```
Comment (jfoder) [inline src/main/java/pl/agh/edu/master_diet/controller/RecentProductController.java:59]: Unnecessary blank line

File: src/main/java/pl/agh/edu/master_diet/core/model/rest/diary/SingleRecentProductResponse.java (hunk 1 above)
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/core/model/rest/diary/SingleRecentProductResponse.java]: Done

File: src/main/java/pl/agh/edu/master_diet/controller/RecentProductController.java (hunk 5 above)
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/controller/RecentProductController.java]: Done

File: src/main/java/pl/agh/edu/master_diet/util/GlobalConstants.java (hunk 4 above)
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/util/GlobalConstants.java]: Done

File: src/main/java/pl/agh/edu/master_diet/service/converter/RecentProductConverter.java (hunk 3 above)
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/service/converter/RecentProductConverter.java:53]: Done

**Pull request merged**